### PR TITLE
[WIP] Mpts stream program number

### PR DESCRIPTION
**Pull request by phunkyfish**
v3.26.0
- Added: Support setting progam number in stream properties for MPTS streams
- Added: Support disabling addon debug logging in debug mode
- Added: Readme updates

### CODE REVIEW

File: src/client.cpp [hunk 1]
```diff
@@ -338,6 +338,33 @@ PVR_ERROR GetChannels(ADDON_HANDLE handle, bool bRadio)
  * Live Streams
  **************************************************************************/
 
+PVR_ERROR GetChannelStreamProperties(const PVR_CHANNEL* channel, PVR_NAMED_VALUE* properties, unsigned int* iPropertiesCount)
+{
+  //
+  // We only use this function to set the program number which comes with every Enigma2 channel. For providers that
+  // use MPTS it allows the FFMPEG Demux to identify the corect Program/PID.
```
Comment (hudokkow) [inline src/client.cpp]: s/corect/correct

sorry for nit picking btw

Comment (phunkyfish) [reply src/client.cpp]: That’s ok!

**Comment by picovidago** [comment]
I'm using kodi 18.2  on windows, how can I test?

**Comment by phunkyfish** [comment]
Are you able to build kodi and the addon?

**Comment by picovidago** [comment]
I tried to build in linux and then take the addon folder to kodi on windows but it does not work

**Comment by picovidago** [comment]
git clone --branch Leia https://github.com/xbmc/xbmc.git
git clone https://github.com/kodi-pvr/pvr.vuplus.git - (mpts-stream-program-number branch)
cd pvr.vuplus && mkdir build && cd build
cmake -DADDONS_TO_BUILD=pvr.vuplus -DADDON_SRC_PREFIX=../.. -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=../../xbmc/build/addons -DPACKAGE_ZIP=1 ../../xbmc/cmake/addons
make

**Comment by phunkyfish** [comment]
You would need to build it on Windows for Windows.

First you would need to build kodi on Windows using the the kodi PR branch, then build the addon on Windows using the addon PR branch.

**Comment by phunkyfish** [comment]
It would be much easier to build everything on Linux and run it there if that’s possible.

**Comment by phunkyfish** [comment]
I’ve never built the addon on Windows so not sure of the steps.

**Comment by picovidago** [comment]
Hello, can you generate and send me a zip so I can test?

**Comment by phunkyfish** [comment]
I don’t currently have a working windows setup to do this. But regardless this would be a case of official builds only I’m afraid.

You would also need to build kodi for Windows as there is a kodi PR too to get this to work.

Here is the kodi PR: https://github.com/xbmc/xbmc/pull/16244

As this is a workaround it may not be accepted and without it there isn’t a way to get this to work.

Sorry I can’t be of more help with this.

Build instructions for Windows is on my todo list for the addon but have to wait until I have a good block of time to work on it.

**Comment by picovidago** [comment]
Hello compiled in linux and I was able to install, I got a version 3.26.0 addon but it still does not open some channels

**Comment by phunkyfish** [comment]
Did you also build xbmc from this branch: https://github.com/phunkyfish/xbmc/tree/pvr-parse-streaminfo?

You need that branch from the accompanying kodi pr:  xbmc/xbmc#16244

With both the addon built and xbmc built it should work.

EDIT: updated branch link.

**Comment by picovidago** [comment]
No, I used https://github.com/xbmc/xbmc.git branch leia.
Let's see if I understand, I have to use the repository https://github.com/phunkyfish/xbmc branch pvr-parse-streaminfo to create the addon and then I have to compile the kodi from this branch and use the addon of this build? Or can I generate the addon and use the regular 18.2 kodi?

**Comment by phunkyfish** [comment]
If you've already built on Leia it might be easier just to make the code change (it's only a single file) and just rerun the final make command again to rebuild kodi (the PR is actually build on Matrix and if you build this you would need to rebuild all of kodi and the addon too).

https://github.com/xbmc/xbmc/pull/16244/files

It's this file: xbmc/cores/VideoPlayer/DVDDemuxers/DVDFactoryDemuxer.cpp

The change is quite small and easy to do by hand.

**Comment by phunkyfish** [comment]
Closing in favour of #233

**Pull request closed without merging**